src/bundler.ts

```diff
@@ -26,28 +26,28 @@ import {ParsedHtmlDocument} from 'polymer-analyzer/lib/html/html-document';
 import {FSUrlLoader} from 'polymer-analyzer/lib/url-loader/fs-url-loader';
 import constants from './constants';
 import * as astUtils from './ast-utils';
+import * as importUtils from './import-utils';
 import * as matchers from './matchers';
 import * as urlUtils from './url-utils';
-import {Bundle, BundleStrategy, AssignedBundle, generateBundles, BundleUrlMapper, BundleManifest, sharedBundleUrlMapper, generateSharedDepsMergeStrategy} from './bundle-manifest';
+import * as bundleManifestLib from './bundle-manifest';
+import {Bundle, BundleStrategy, AssignedBundle, BundleUrlMapper, BundleManifest} from './bundle-manifest';
 import {BundledDocument, DocumentCollection} from './document-collection';
-import {buildDepsIndex} from './deps-index';
+import * as depsIndexLib from './deps-index';
 import {UrlString} from './url-utils';
 
-// TODO(usergenic): Document every one of these options.
+// TODO(garlicnation): resolve <base> tags.
+// TODO(garlicnation): find transitive dependencies of specified excluded files.
+// TODO(garlicnation): ignore <link> in <template>
+// TODO(garlicnation): Add noopResolver for external urls.
+// TODO(garlicnation): Add noopResolver for excluded urls.
+// TODO(garlicnation): Add redirectResolver for fakeprotocol:// urls
+// TODO(usergenic): Add plylog
 export interface Options {
   // When provided, relative paths will be converted to absolute paths where
   // `basePath` is the root url.  This path is equal to the folder of the
   // bundled url document of the analyzer.
-  //
-  // TODO(usergenic): If multiple `bundle()` calls are made `basePath` can
-  // produce incompatile absolute paths if the same `basePath` is used for
-  // `bundle()` targets in different folders.  Possible solutions include
-  // removing basePath behavior altogether or supplementing it with a `rootPath`
-  // or other hint to fix the top-level folder.
   basePath?: string;
```

Comment:
@justinfagnani Are you saying that there's another something down the toolchain which can prefix all the paths in all the the things and we can pull this out of bundler altogether because I WOULD LOVE THAT.